sphinxcontrib/datatemplates/directive.py

```diff
@@ -127,10 +127,18 @@ def _load_data(self, resolved_path):
 
 
 class DataTemplateYAML(DataTemplateWithEncoding):
+    option_spec = dict(DataTemplateBase.option_spec, **{
+        'multiple-documents': rst.directives.flag,
+    })
+
     def _load_data(self, resolved_path):
         with open(resolved_path,
                   'r',
                   encoding=self.options.get('encoding', 'utf-8-sig')) as f:
+            if 'multiple-documents' in self.options:
+                return list(
+                    yaml.safe_load_all(f)
+                )  # force loading all documents now so the file can be closed
```

Comment:
I think loading everything is the right approach. It also ensures that someone could use len(data) in their template.